runtime-sdk/modules/evm/src/precompile/auxiliary.rs

```diff
@@ -0,0 +1,44 @@
+//! Implements additional precompiles that are not part of the EVM specification.
+use evm::{
+    executor::stack::{PrecompileHandle, PrecompileOutput},
+    ExitSucceed,
+};
+use ripemd160::Digest as _;
+use sha2::Sha512Trunc256;
+
+use super::{record_linear_cost, PrecompileResult};
+
+pub(super) fn call_sha512_256(handle: &mut impl PrecompileHandle) -> PrecompileResult {
+    record_linear_cost(handle, handle.input().len() as u64, 60, 12)?; // TODO: sensible gas costs, not sure what to set.
+
+    let mut hasher = Sha512Trunc256::new();
+    hasher.update(handle.input());
+    let digest = hasher.finalize();
+
+    Ok(PrecompileOutput {
+        exit_status: ExitSucceed::Returned,
+        output: digest.to_vec(),
+    })
+}
+
+#[cfg(test)]
+mod test {
+    use super::super::test::*;
+
+    #[test]
+    fn test_sha512_256() {
+        let input = "38d18acb67d25c8bb9942764b62f18e17054f66a817bd4295423adf9ed98873e000000000000000000000000000000000000000000000000000000000000001b38d18acb67d25c8bb9942764b62f18e17054f66a817bd4295423adf9ed98873e789d1dd423d25f0772d2748d60f7e4b81bb14d086eba8e8e8efb6dcff8a4ae02";
```

Comment:
Would probably be nice to have E2E test in an actual .sol contract for all/most precompiles?